.github/workflows/main.yml

```diff
@@ -1,41 +1,64 @@
 name: CI
 
-on: [push, pull_request]
+on: [ push, pull_request ]
+
+permissions:
+  contents: read
 
 jobs:
   lint-pr:
     if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
-      - uses: TrueBrain/actions-flake8@v1.4.1
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - uses: TrueBrain/actions-flake8@9a43ff1b2c7b96f3edffc48a49973ce3de116ba1
+        # mirrored at https://github.com/mitmproxy/mitmproxy/settings/actions
```

Comment:
We now pin all non-GitHub actions to a specific revision, and also maintain a whitelist at https://github.com/mitmproxy/mitmproxy/settings/actions.

Reply:
This action is only for convenience to speed up the dev cycle? or does it provide anything we don't have in tox?